cybersecurity/port-scanner/devashmit-day12/port_scanner.py

```
@@ -0,0 +1,39 @@
+def main():
+    print("\n🔐 Multi-threaded Port Scanner\n   Only scan systems you own or have permission to scan.\n")
+    target=input("Target IP: ").strip()
+    try: ip=socket.gethostbyname(target)
+    except: print("Cannot resolve host"); return
+    try: start=int(input("Start port: ")); end=int(input("End port:   "))
+    except: print("Invalid ports"); return
+
+    print(f"\n  Scanning {ip} ports {start}-{end}...\n")
+    ts=[]; 
+    for p in range(start,end+1):
+        t=threading.Thread(target=scan,args=(ip,p),daemon=True); ts.append(t); t.start()
```

Ruff (0.15.15)

[error] Multiple statements on one line (semicolon)

(E702)